### ISSUE TITLE
Support loading instantiated data structures into the `MerkleStore`

### ISSUE THREAD

**Issue by frisitano** [0xPolygonMiden/crypto#136]
As it stands there is not way to load an instantiated data structure (`Mmr`, `Smt`, `MerkleTree`) into the `MerkleStore`.  The `MerkleStore` API expects iterators of a specific format depending on the structure being loaded.  We should implement methods on `Mmr`, `Smt`, `MerkleTree` that produce the required iterators that will allow the instantiated data structures to be loaded into the `MerkleStore`.

**Comment by hackaugusto** [comment]
My opinion is that we should make a trait for [`inner_nodes`](https://github.com/0xPolygonMiden/crypto/blob/b5eb68e46c87d13406f2f67d1c8980d7408d24b8/src/merkle/merkle_tree.rs#L157) iterator, and implement a generic `From` trait for that.

**Comment by bobbinth** [comment]
Agreed - we can improve the API quite a bit. I like @hackaugusto's suggestion of having an iterator over inner nodes - though, I think we should have a dedicated method instead of (or maybe in addition to) implementing `From`. This method could look something like:
```Rust
pub fn append<R, I>(&mut self, nodes: R) -> &mut Self
where
    R: IntoIterator<IntoIter = I>,
    I: Iterator<Item = InnerNodeInfo> + ExactSizeIterator,
{
    for node in nodes.into_iter() {
        debug_assert_eq!(
            Rpo256::merge(&[node.left.into(), node.right.into()]),
            node.value.into()
        );

        self.nodes.insert(
            node.value.into(),
            Node {
                left: node.left.into(),
                right: node.right.into(),
            },
        );
    }

    self
}
```
If we do this, we can probably get rid of most `with_*` and `add_*` methods.

**Comment by bobbinth** [comment]
Closed by #139.
